### DISCUSSION TITLE
C++ library and examples for the Haxophone

### DISCUSSION THREAD

**Discussion by StaffanMelin** [cardonabits/haxo-rs#43] [Show and tell]
Hi all,

I've started to create a C++ framework for making sounds with the Haxophone.

It is now starting to be playable, but I have loads of ideas for how this is going to move forward!

Check out the short demo soundfile, the code and the issues at
[haxo-cs](https://github.com/StaffanMelin/haxo-cs)

I also have some ideas for hardware mods, esp a foot controller/expression pedal, that I will soon start to work on.

But first I will add more examples, showing what the Haxo and my lib can do together. I hope you will like it! :)

Last but not least, a huge thank you to Javier for making the haxo so beautiful and open!

Staffan
SWEDEN

**Comment by jcard0na** [comment]
Nice project!  I just listened to your demo audio and it's very soothing and futuristic at the same time.  Thanks for sharing and the kind words!

**Reply by StaffanMelin** [reply]
Thank you @jcard0na! I've started to learn some Rust, but moving my audio code to Rust is right now just too big a project.

**Comment by Ekolide** [comment]
Amazing! If you need someone to help test, I'm more than willing to help. There's been ideas in my mind to implement something like [AMY](https://github.com/bwhitman/amy) as a sound source, add the unused buttons/keyswitches or pitch bend or a sort of bite sensor, or move the Haxophone os to something like buildroot. Best of luck to you I'm excited to see what you come up with!

**Reply by StaffanMelin** [reply]
Thank you @Ekolide! I had not heard of AMY before, but it seems very much like my rtDStudio project (which began as an Open Frameworks project, demo here of what it can do: https://www.youtube.com/watch?v=65uiYlsG29Q&list=PLkncOj8OBYS9b_CTPLYD4HGQsUDeA2UKo&index=14).

I am definitely going to use the unused keys. :)

I'll get back to you when I have something to test, hopefully in about a month.

Thank you for your feedback!

**Comment by StaffanMelin** [comment]
So this took longer than expected, but now DStudio 2.0 is out, with support for the Haxophone! It also includes instructions on how to build a foot controller for the Haxophone.

https://github.com/StaffanMelin/dstudio/tree/main

**Comment by jcard0na** [comment]
Great!  Thanks for sharing!
